File: src/main/java/fi/finlit/edith/sql/domain/NameForm.java
package fi.finlit.edith.sql.domain;

import javax.persistence.Embeddable;

@Embeddable
public class NameForm {
    private String description;

    private String first;

    private String last;

    public NameForm() {
    }

    public NameForm(String name, String description) {
        last = name;
        this.description = description;
    }

    public NameForm(String first, String last, String description) {
        this.first = first;
        this.last = last;
        this.description = description;
    }

    public String getDescription() {
        return description;
    }

    public String getName() {
        StringBuilder builder = new StringBuilder();
        if (first != null) {
            builder.append(first);
        }
        if (first != null && last != null) {
            builder.append(" ");
        }
        if (last != null) {
            builder.append(last);
        }
        return builder.toString();
    }

    public void setName(String name) {
        last = name;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getFirst() {
        return first;
    }

    public String getLast() {
        return last;
    }

    public void setLast(String last) {
        this.last = last;
    }

    public void setFirst(String first) {
        this.first = first;
    }

    @Override
    public String toString() {
        return "NameForm [description=" + description + ", name=" + getName() + "]";
    }
}

File: src/main/scala/fi/finlit/edith/sql/domain/NameForm.scala
package fi.finlit.edith.sql.domain

import javax.persistence.Embeddable
import org.apache.commons.lang.StringUtils
import scala.reflect.BeanProperty
//remove if not needed
import scala.collection.JavaConversions._

@Embeddable
class NameForm() {

  @BeanProperty
  var description: String = _

  @BeanProperty
  var first: String = _

  @BeanProperty
  var last: String = _

  def this(name: String, description: String) {
    last = name
    this.description = description
  }

  def this(first: String, last: String, description: String) {
    this.first = first
    this.last = last
    this.description = description
  }

  def getName(): String = {
    var builder = new StringBuilder()
    if (first != null) {
      builder.append(first)
    }
    if (first != null && last != null) {
      builder.append(" ")
    }
    if (last != null) {
      builder.append(last)
    }
    builder.toString
  }

  def isValid(): Boolean = {
    StringUtils.isNotBlank(first) || StringUtils.isNotBlank(last)
  }

  override def toString(): String = {
    "NameForm [description=" + description + ", name=" + getName + "]"
  }
}
